lib/geoengineer/utils/aws_clients.rb

```diff
@@ -11,80 +11,130 @@ def self.stubbed?
     @stub_aws || false
   end
 
+  def self.client_params(provider = nil)
+    client_params = { stub_responses: stubbed? }
+    client_params[:region] = provider.region if provider
+    client_params
+  end
+
+  def self.client_cache(client, provider, &block)
+    @client_cache ||= {}
+    key = "#{client}_" + (provider&.terraform_id || GeoEngineer::Resource::DEFAULT_PROVIDER)
+    @client_cache[key] ||= block.call
+  end
+
   # Clients
-  def self.cloudwatch
-    @aws_cloudwatch ||= Aws::CloudWatch::Client.new({ stub_responses: stubbed? })
+  def self.cloudwatch(provider = nil)
+    self.client_cache('cloudwatch', provider) {
+      Aws::CloudWatch::Client.new(client_params(provider))
```

Comment:
__method__ in the client_cache method will equal `:client_cache` so the outer method must be added as well. You could instantiate a new client with each call. This might make it a bit clearer.